### PR TITLE
ShuffleLog: NetworkTables tool for 2023

### PR DESCRIPTION
**Pull request by rmheuer**

### CODE REVIEW

**Review by foehammer127 — COMMENTED** [review]
👍

**Comment by foehammer127** [comment]
Change/Resolve these and then we are good.

File: ShuffleLog/build.gradle [hunk 1]
```diff
@@ -19,11 +21,26 @@ compileJava {
     targetCompatibility = '11'
 }
 
-def WPILIB_VERSION = '2023.2.1'
+def WPILIB_VERSION = wpi.versions.wpilibVersion.get()
```
[the remark this answers is not in the document]
Comment (rmheuer) [reply ShuffleLog/build.gradle:24]: No, the version number is provided as the version for the GradleRIO plugin

File: ShuffleLog/src/main/java/com/swrobotics/shufflelog/tool/data/nt/NetworkTableRepr.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        for (NetworkTableRepr repr : toRemove.values())
+            repr.close();
+
+        return out;
+    }
+
+    public Set<NetworkTableValueRepr> getValues() {
+        Set<NetworkTableValueRepr> out = new HashSet<>();
+
+        Map<String, NetworkTableValueRepr> toRemove = new HashMap<>(valueCache);
```
[the remark this answers is not in the document]
Comment (rmheuer) [reply ShuffleLog/src/main/java/com/swrobotics/shufflelog/tool/data/nt/NetworkTableRepr.java:96]: The Map interface works fine here and is best practice (Effective Java 52: Refer to objects by their interface).

[the remark this answers is not in the document]
Comment (foehammer127) [reply ShuffleLog/src/main/java/com/swrobotics/shufflelog/tool/data/nt/NetworkTableRepr.java:96]: Its Effective Java 64 not 52

**Pull request merged**